compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/NodeFinder.java

```diff
@@ -806,7 +806,11 @@ public void visit(BLangTernaryExpr ternaryExpr) {
 
     @Override
     public void visit(BLangWaitExpr awaitExpr) {
-        lookupNodes(awaitExpr.exprList);
+        if (lookupNodes(awaitExpr.exprList)) {
+            return;
+        }
+
+        setEnclosingNode(awaitExpr, awaitExpr.pos);
```

Comment:
Since latest nodeFInder changes reverted in the master (https://github.com/ballerina-platform/ballerina-lang/pull/38304), Shall we check this change still valid after the revert ?